pkg/storage/replica_command.go

```diff
@@ -1565,6 +1565,8 @@ func evalPushTxn(
 	if reply.PusheeTxn.Epoch < args.PusheeTxn.Epoch {
 		reply.PusheeTxn.Epoch = args.PusheeTxn.Epoch
 	}
+	reply.PusheeTxn.UpgradePriority(args.PusheeTxn.Priority)
```

Comment:
I think we should simply call `reply.PusheeTxn.Update(args.PusheeTxn)` and remove the calls above here to forward the timestamp and epoch.

You can also remove the `TODO(kaneda) Update Sequence as well?`, as that will be handled appropriately.

It does make sense to move the call to `Transaction.Update()` from here below the two early-exit clauses just below this comment for already-committed/aborted and the timestamp already-pushed-far-enough.